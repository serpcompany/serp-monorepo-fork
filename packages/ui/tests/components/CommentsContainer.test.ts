import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { afterAll, describe, expect, it, vi } from 'vitest';
import { ref } from 'vue';
// Import mocks first so they're available when components are imported
import { cleanupDateMocks } from '../mockDateImports';
import '../mockUseUserSession';
// Then import components
import CommentsContainer from '../../components/CommentsContainer.vue';
import ComponentRender from '../componentRender';

// Mock composables used in CommentsContainer.vue
mockNuxtImport('useToast', () => () => ({
  add: vi.fn()
}));

mockNuxtImport('useFetch', () => async () => ({
  data: { value: { message: 'success', id: 1234 } },
  error: { value: null }
}));

mockNuxtImport('useRequestHeaders', () => () => ({}));

// Define various scenarios to snapshot test different prop combinations.
const scenarios: [string, { props: Record<string, unknown> }][] = [
  [
    'default state with no comments',
    {
      props: {
        module: 'blog',
        id: 1,
        // Using default values for other props.
        comments: []
      }
    }
  ],
  [
    'with a single comment',
    {
      props: {
        module: 'blog',
        id: 1,
        comments: [
          {
            id: 101,
            email: 'user@example.com',
            name: 'John Doe',
            image: 'https://example.com/avatar.jpg',
            content: 'Great post!',
            createdAt: '1973-04-29T21:33:09+00:00',
            updatedAt: '1973-04-29T21:33:09+00:00',
            replies: []
          }
        ]
      }
    }
  ],
  [
    'with nested comment',
    {
      props: {
        module: 'blog',
        id: 1,
        comments: [
          {
            id: 204,
            email: 'user@example.com',
            name: 'John Doe',
            image: 'https://example.com/avatar.jpg',
            content: 'Great post!',
            createdAt: '1973-04-29T21:33:09+00:00',
            updatedAt: '1973-04-29T21:33:09+00:00',
            replies: [
              {
                id: 205,
                email: 'user1@example.com',
                name: 'Alice',
                image: 'https://example.com/alice.jpg',
                content: 'Nice work!',
                createdAt: '1973-04-29T21:33:09+00:00',
                updatedAt: '1973-04-29T21:33:09+00:00',
                replies: [
                  {
                    id: 206,
                    email: 'user1@example.com',
                    name: 'Alice',
                    image: 'https://example.com/alice.jpg',
                    content: 'Nice work!',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: []
                  },
                  {
                    id: 207,
                    email: 'user1@example.com',
                    name: 'Alice',
                    image: 'https://example.com/alice.jpg',
                    content: 'Nice work!',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: []
                  }
                ]
              }
            ]
          }
        ]
      }
    }
  ],
  [
    'with multiple comments and custom styling',
    {
      props: {
        module: 'forum',
        id: 2,
        background: '#f0f0f0',
        commentBackgroundColor: '#fff',
        commentTextColor: '#333',
        userNameColor: '#ff0000',
        comments: [
          {
            id: 201,
            email: 'user1@example.com',
            name: 'Alice',
            image: 'https://example.com/alice.jpg',
            content: 'Nice work!',
            createdAt: '1973-04-29T21:33:09+00:00',
            updatedAt: '1973-04-29T21:33:09+00:00',
            replies: []
          },
          {
            id: 202,
            email: 'user2@example.com',
            name: 'Bob',
            image: 'https://example.com/bob.jpg',
            content: 'I disagree',
            createdAt: '1973-04-29T21:33:09+00:00',
            updatedAt: '1973-04-29T21:33:09+00:00',
            replies: []
          }
        ]
      }
    }
  ]
];

describe('CommentsContainer Snapshot', () => {
  it.each(scenarios)('renders %s correctly', async (desc: string, options) => {
    // Render the component with the provided props using the Nuxt testing utility.
    const html = await ComponentRender(
      `CommentsContainer ${desc}`,
      options,
      CommentsContainer
    );
    expect(html).toMatchSnapshot();
  });

  afterAll(() => {
    cleanupDateMocks();
  });
});
