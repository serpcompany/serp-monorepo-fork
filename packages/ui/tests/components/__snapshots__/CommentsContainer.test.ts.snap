// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentsContainer Snapshot > renders default state with no comments correctly 1`] = `
"<div data-v-e7198dba="" class="comments" style="background: transparent;">
  <div data-v-e7198dba="" class="innerWrapper">
    <div data-v-e7198dba="" class="userName">
      <div data-v-e7198dba="" class="addComment">
        <div data-v-e7198dba="" class="w-12 rounded-t-full">
          <!--v-if-->
        </div>
        <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
    <!--v-if-->
  </div>
</div>"
`;

exports[`CommentsContainer Snapshot > renders with a single comment correctly 1`] = `
"<div data-v-e7198dba="" class="comments" style="background: transparent;">
  <div data-v-e7198dba="" class="innerWrapper">
    <div data-v-e7198dba="" class="userName">
      <div data-v-e7198dba="" class="addComment">
        <div data-v-e7198dba="" class="w-12 rounded-t-full">
          <!--v-if-->
        </div>
        <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true">
      <div data-v-d830712d="" data-v-e7198dba="" class="comment-wrapper">
        <div data-v-d830712d="" class="wrapper">
          <div data-v-d830712d="" class="w-12 rounded-t-full">
            <!---->
          </div>
          <div data-v-d830712d="" class="comment-wrapper">
            <div data-v-d830712d="" class="name-wrapper">
              <div data-v-d830712d="" class="name text-(--ui-primary)">John Doe</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
              <!--v-if-->
              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
            </div>
            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Great post!</div>
            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
              <div data-v-d830712d="" class="user-name text-(--ui-primary)">John Doe</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
            </div>
            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
              <div data-v-d830712d="">Never mind</div>
            </div>
            <div data-v-d830712d="" class="reply">
              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
              <!--v-if--><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="delete">
                <div data-v-d830712d="" class="delete-text">Delete</div>
                <!--v-if-->
              </div>
            </div>
            <!--v-if-->
            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </transition-group-stub>
    <!--v-if-->
  </div>
</div>"
`;

exports[`CommentsContainer Snapshot > renders with multiple comments and custom styling correctly 1`] = `
"<div data-v-e7198dba="" class="comments" style="background: #f0f0f0;">
  <div data-v-e7198dba="" class="innerWrapper">
    <div data-v-e7198dba="" class="userName">
      <div data-v-e7198dba="" class="addComment">
        <div data-v-e7198dba="" class="w-12 rounded-t-full">
          <!--v-if-->
        </div>
        <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true">
      <div data-v-d830712d="" data-v-e7198dba="" class="comment-wrapper">
        <div data-v-d830712d="" class="wrapper">
          <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/alice.jpg 1x, https://example.com/alice.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/alice.jpg"></div>
          <div data-v-d830712d="" class="comment-wrapper">
            <div data-v-d830712d="" class="name-wrapper">
              <div data-v-d830712d="" class="name text-(--ui-primary)">Alice</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
              <!--v-if-->
              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
            </div>
            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Nice work!</div>
            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
              <div data-v-d830712d="" class="user-name text-(--ui-primary)">Alice</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
            </div>
            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
              <div data-v-d830712d="">Never mind</div>
            </div>
            <div data-v-d830712d="" class="reply">
              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
              <!--v-if--><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="delete">
                <div data-v-d830712d="" class="delete-text">Delete</div>
                <!--v-if-->
              </div>
            </div>
            <!--v-if-->
            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
      <div data-v-d830712d="" data-v-e7198dba="" class="comment-wrapper">
        <div data-v-d830712d="" class="wrapper">
          <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/bob.jpg 1x, https://example.com/bob.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/bob.jpg"></div>
          <div data-v-d830712d="" class="comment-wrapper">
            <div data-v-d830712d="" class="name-wrapper">
              <div data-v-d830712d="" class="name text-(--ui-primary)">Bob</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
              <!--v-if-->
              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
            </div>
            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">I disagree</div>
            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
              <div data-v-d830712d="" class="user-name text-(--ui-primary)">Bob</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
            </div>
            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
              <div data-v-d830712d="">Never mind</div>
            </div>
            <div data-v-d830712d="" class="reply">
              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
              <!--v-if--><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="delete">
                <div data-v-d830712d="" class="delete-text">Delete</div>
                <!--v-if-->
              </div>
            </div>
            <!--v-if-->
            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </transition-group-stub>
    <!--v-if-->
  </div>
</div>"
`;

exports[`CommentsContainer Snapshot > renders with nested comment correctly 1`] = `
"<div data-v-e7198dba="" class="comments" style="background: transparent;">
  <div data-v-e7198dba="" class="innerWrapper">
    <div data-v-e7198dba="" class="userName">
      <div data-v-e7198dba="" class="addComment">
        <div data-v-e7198dba="" class="w-12 rounded-t-full">
          <!--v-if-->
        </div>
        <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true">
      <div data-v-d830712d="" data-v-e7198dba="" class="comment-wrapper">
        <div data-v-d830712d="" class="wrapper">
          <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/avatar.jpg"></div>
          <div data-v-d830712d="" class="comment-wrapper">
            <div data-v-d830712d="" class="name-wrapper">
              <div data-v-d830712d="" class="name text-(--ui-primary)">John Doe</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
              <!--v-if-->
              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
            </div>
            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Great post!</div>
            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
              <div data-v-d830712d="" class="user-name text-(--ui-primary)">John Doe</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
            </div>
            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
              <div data-v-d830712d="">Never mind</div>
            </div>
            <div data-v-d830712d="" class="reply">
              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Hide replies</div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
              <div data-v-d830712d="" class="delete">
                <div data-v-d830712d="" class="delete-text">Delete</div>
                <!--v-if-->
              </div>
            </div>
            <!--v-if-->
            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true">
              <div data-v-d830712d="" class="comment-wrapper">
                <div data-v-d830712d="" class="wrapper">
                  <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/alice.jpg 1x, https://example.com/alice.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/alice.jpg"></div>
                  <div data-v-d830712d="" class="comment-wrapper">
                    <div data-v-d830712d="" class="name-wrapper">
                      <div data-v-d830712d="" class="name text-(--ui-primary)">Alice</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                      <!--v-if-->
                      <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
                    </div>
                    <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Nice work!</div>
                    <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                      <div data-v-d830712d="" class="user-name text-(--ui-primary)">Alice</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                      <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
                    </div>
                    <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                      <div data-v-d830712d="">Never mind</div>
                    </div>
                    <div data-v-d830712d="" class="reply">
                      <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div><span data-v-d830712d="" class="dot">•</span>
                      <div data-v-d830712d="" class="hover:text-(--ui-primary)">Hide replies</div><span data-v-d830712d="" class="dot">•</span>
                      <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                      <div data-v-d830712d="" class="delete">
                        <div data-v-d830712d="" class="delete-text">Delete</div>
                        <!--v-if-->
                      </div>
                    </div>
                    <!--v-if-->
                    <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true">
                      <div data-v-d830712d="" class="comment-wrapper">
                        <div data-v-d830712d="" class="wrapper">
                          <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/alice.jpg 1x, https://example.com/alice.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/alice.jpg"></div>
                          <div data-v-d830712d="" class="comment-wrapper">
                            <div data-v-d830712d="" class="name-wrapper">
                              <div data-v-d830712d="" class="name text-(--ui-primary)">Alice</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                              <!--v-if-->
                              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
                            </div>
                            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Nice work!</div>
                            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                              <div data-v-d830712d="" class="user-name text-(--ui-primary)">Alice</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
                            </div>
                            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                              <div data-v-d830712d="">Never mind</div>
                            </div>
                            <div data-v-d830712d="" class="reply">
                              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
                              <!--v-if--><span data-v-d830712d="" class="dot">•</span>
                              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                              <div data-v-d830712d="" class="delete">
                                <div data-v-d830712d="" class="delete-text">Delete</div>
                                <!--v-if-->
                              </div>
                            </div>
                            <!--v-if-->
                            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
                            <!--v-if-->
                          </div>
                        </div>
                      </div>
                      <div data-v-d830712d="" class="comment-wrapper">
                        <div data-v-d830712d="" class="wrapper">
                          <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/alice.jpg 1x, https://example.com/alice.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/alice.jpg"></div>
                          <div data-v-d830712d="" class="comment-wrapper">
                            <div data-v-d830712d="" class="name-wrapper">
                              <div data-v-d830712d="" class="name text-(--ui-primary)">Alice</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                              <!--v-if-->
                              <div data-v-d830712d="" class="expand hover:text-(--ui-primary)"><span data-v-d830712d="" title="Hide Comment">−</span></div>
                            </div>
                            <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">Nice work!</div>
                            <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                              <div data-v-d830712d="" class="user-name text-(--ui-primary)">Alice</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                              <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
                            </div>
                            <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                              <div data-v-d830712d="">Never mind</div>
                            </div>
                            <div data-v-d830712d="" class="reply">
                              <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
                              <!--v-if--><span data-v-d830712d="" class="dot">•</span>
                              <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                              <div data-v-d830712d="" class="delete">
                                <div data-v-d830712d="" class="delete-text">Delete</div>
                                <!--v-if-->
                              </div>
                            </div>
                            <!--v-if-->
                            <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
                            <!--v-if-->
                          </div>
                        </div>
                      </div>
                    </transition-group-stub>
                    <!--v-if-->
                  </div>
                </div>
              </div>
            </transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </transition-group-stub>
    <!--v-if-->
  </div>
</div>"
`;
